ci/run_easymesh_cert.sh

```diff
@@ -0,0 +1,115 @@
+}
+
+usage() {
+    echo "usage: $(basename $0) [-hboev] <test> [test]"
+    echo "  options:"
+    echo "      -h|--help - display this help."
+    echo "      -b|--branch - prplmesh branch to use (default master latest)"
+    echo "      -o|--log-folder - path to put the logs to (make sure it does not contain previous logs)."
+    echo "      -e|--easymesh-cert - path to easymesh_cert repository (default ../easymesh_cert)"
+    echo "      -u|--upload - upload results"
+    echo "      -v|--verbose - set verbosity (ucc logs also redirected to stdout)"
+    echo "'test' can either be a test name, or the path to a file containing"
+    echo " test names (newline-separated)."
+    echo
+    echo "Multiple test files/names are allowed, and they can also be mixed."
+    echo "For example, the following call would run the test 'MAP-5.3.1', and all the tests"
+    echo "contained in the 'tests/all_controller_test.txt' file:"
+    echo "$(basename "$0") MAP-5.3.1 tests/all_controller_test.txt"
+    echo ""
+    echo "The default is to run all agent certification tests."
+}
+
+main() {
+     OPTS=$(getopt -o 'hb:o:e:u:v' --long help,branch:,log-folder:,easymesh-cert:,upload:,verbose -n 'parse-options' -- "$@")
+
+    if [ $? != 0 ] ; then echo "Failed parsing options." >&2 ; usage; exit 1 ; fi
+
+    eval set -- "$OPTS"
+
+    while true; do
+        case "$1" in
+            -h | --help)            usage; exit 0;;
+            -b | --branch)          BRANCH="$2"; shift 2;;
+            -o | --log-folder)      LOG_FOLDER="$2"; shift 2;;
+            -v | --verbose)         VERBOSE=true; VERBOSE_OPT="-v"; shift;;
+            -e | --easymesh-cert)   EASYMESH_CERT_PATH="$2"; shift 2;;
+            -u | --upload)          UPLOAD_PATH="$2"; shift 2;;
+            -- ) shift; break ;;
+            * ) err "unsupported argument $1"; usage; exit 1;;
+        esac
+    done
+
+    TESTS="$@"
+
+    [ -n "$TESTS" ] || TESTS="$EASYMESH_CERT_PATH/tests/all_agent_tests.txt"
+    [ -n "$LOG_FOLDER" ] || LOG_FOLDER="$EASYMESH_CERT_PATH/logs/$(date +%F_%H-%M-%S)"
+    info "Logs location: $LOG_FOLDER"
+    info "Tests to run: $TESTS"
+
+    info "download latest ipk"
+    run "$PRPLMESH_PATH"/tools/download_ipk.sh --branch "$BRANCH"
+
+    info "deploy latest ipk"
+    run "$PRPLMESH_PATH"/tools/deploy_ipk.sh "$USER"@"$IPADDR" "$IPK"
+
+    info "Start running tests"
+    run "$EASYMESH_CERT_PATH"/run_test_file.sh "$VERBOSE_OPT" -o "$LOG_FOLDER" -d "$TARGET_DEVICE" "$TESTS"
+
+    passed=$(ls $LOG_FOLDER/PASS | wc -l)
+    failed=$(ls $LOG_FOLDER/FAIL | wc -l)
+    total=$((failed+passed))
+    if [ "$VERBOSE" = "true" ]; then
+        info "Passed tests:"
+        ls "$LOG_FOLDER/PASS"
+        info "Failed tests:"
+        ls "$LOG_FOLDER/FAIL"
+    fi
+    info "Passed $passed/$total"
+    info "Uploading results to $UPLOAD_PATH not supported yet"
+    # TODO upload results
+    info "done"
+}
+
+USER=root
+BRANCH=master
+IPADDR=192.168.250.170
```

Comment:
I really don't like that the IP address is hardcoded in the script, but there's not really an alternative I guess. Except, maybe it would make sense to default it to `"${TARGET_DEVICE}-1"`.

Reply:
So we count on ~/.ssh/config. OK, make sense I guess.